length_of_path.py

```diff
@@ -0,0 +1,117 @@
+from pathlib import Path
+import csv
+import math
+import functools as func
+
+
+class IInterface(object):
+    """Interface class with usable methods"""
+    def read(self):
+        """read from some file"""
+
+class CSVReader(IInterface):
+    
+    def __eq__(self, value):
+        if value.isdigit()==True:
+            pass
+        else:
+            print (value)
+            raise Exception("One or more wrong coordinates")
+
+    def validation(self,row):
+        for i in row:
+            self.__eq__(i)
+
+    def read(self, file_name):
+        """
+        :param file_name: include file location and file name
+        :return: array of coordinates from file
+        """
+        coordinates=[]  
+        with open(file_name) as csv_file:
+            csv_reader = csv.reader(csv_file, delimiter=',')
+            line_count = 0
+            for row in csv_reader:
+                if line_count==0:
+                    line_count+=1
+                else:
+                    self.validation(row)
+                    coordinates.append(([row[0]])+[row[1]])  #gives us array list
+                    line_count+=1
+        return coordinates
+
+
+class ReadersFactory (IInterface):
+    """ 
+    Can choose Reader for file;
+    :param file_name: include file location and file name
+    :return: array of coordinates from file
+    """
+
+    def read(self,file_name):
+        readers_dict={ '.csv': CSVReader()
```

Comment:
define this dict outside of this class, and call values of it throw *some_dict[key]*

Reply:
so you will have constant time complexity, not linear